### PR TITLE
chore(events): add events related documentation

### PR DESCRIPTION
**Pull request by Richard-lumapps**
CF-144

Add events related documentation
Add search events documentation
Migrate navigation and widget rendered documentation to event document
Made `onNavigation` and `onWidgetRendered` functions deprecated

### CODE REVIEW

File: docs/javascript/api.md [hunk 1]
```diff
@@ -33,8 +33,9 @@ Where `callback` is a function that will receive a set of `parameters` and retur
 | `constants`    | A group of variables that hold the different [constants](#constants) that can be used for creating components on your site.                                        |
 | `render`       | Function that allows rendering a component in a specific placement and target. See more details for this function [here](#render).                                 |
 | `session`      | Object that contains several values from the current session that can be useful when creating customizations. See more details for this function [here](#session). |
-| `onNavigation` | Function that will be executed once any widget in a content is rendered. See more details for this function [here](#on-widget-rendered)                          |
-| `onWidgetRendered` | Function that will be executed once the application has performed a navigation. See more details for this function [here](#on-navigation)                          |
+| `on`  | Function that will be executed upon user actions or component rendrering. See more details for this function [here](#on)                          |
```
Comment (juanigalan91) [inline docs/javascript/api.md]: Small typo

```suggestion
| `on`  | Function that will be executed upon user actions or component rendering. See more details for this function [here](#on)                          |
```

File: docs/javascript/api.md [hunk 2]
```diff
@@ -44,6 +45,19 @@ And `configuration` is an object that allows these properties:
...
+| `events.NAVIGATION`                    | Event id for user navigation events.                                                           | [Documentation](./capabilities#event-navigation)            |
+| `events.SEARCH_BOX`              | Event id for the user searchbox interaction events.                                  | [Documentation](./capabilities#event-searchbox)              |
+| `events.SEARCH_FILTER`              | Event id for the user filter interaction events on the search page.                                                   | [Documentation](./capabilities#event-search-filter)              |
+| `events.SEARCH_RESULT`                | Event id for the search result events triggered when user makes a search query.                                                      | [Documentation](./capabilities#event-search-result)                |
+| `events.SEARCH_SORT`     | Event id for the user sort interaction events on the search page. | [Documentation](./capabilities#event-search-sort)     |
```
Comment (juanigalan91) [inline docs/javascript/api.md]: To be updated by just  `events.SEARCH`

File: docs/javascript/api.md [hunk 3]
```diff
@@ -1185,22 +1213,53 @@ window.lumapps.customize(({ onNavigation }) => {
 });
 ```
 
-`onNavigation` is a function that will be called on each navigation. It receives the following parameters:
-- `currentPage`: Id of the current page.
+`onNavigation` is a function that will be called on each navigation. **This function is now deprecated**. Please `on` function with the [Navigation event](./api#eventsnavigation) instead.
```
Comment (juanigalan91) [inline docs/javascript/api.md]: Can we move this just below the `### on navigation` and add an `IMPORTANT` prefix in bold?

File: docs/javascript/api.md (hunk 1 above)
Comment (Richard-lumapps) [reply docs/javascript/api.md]: Fixed!

File: docs/javascript/api.md (hunk 2 above)
Comment (Richard-lumapps) [reply docs/javascript/api.md]: Done

File: docs/javascript/api.md (hunk 3 above)
Comment (Richard-lumapps) [reply docs/javascript/api.md]: Done as well

**Review by juanigalan91 — APPROVED** [review]

**Pull request merged**